R/population-study-design.R

```diff
@@ -0,0 +1,226 @@
+#' @title addStudyParameters
+#' @param population `Population` object
+#' @param studyDesignFile file name of study design table
+#' @export
+#' @import ospsuite
+addStudyParameters <- function(population, studyDesignFile = NULL) {
+  validateIsOfType(population, "Population")
+  validateIsString(studyDesignFile, nullAllowed = TRUE)
+  if (is.null(studyDesignFile)) {
+    return(invisible())
+  }
+  studyDesign <- loadStudyDesign(studyDesignFile)
+
+  initialTargetValues <- rep(NA, population$count)
+  populationData <- ospsuite::populationAsDataFrame(population)
+
+  for (target in studyDesign$targets) {
+    parameterPath <- target$name
+
+    updatedTargetValues <- updateTargetValues(initialTargetValues, target$values, studyDesign$source, populationData)
+
+    population$setParameterValues(parameterPath, updatedTargetValues)
+  }
+}
+
+#' @title updateTargetValues
+#' @param values initial vector to update
+#' @param targetValues vector of values to be assigned using `sourceExpressions`
+#' @param sourceExpressions study design expressions to be evaluated
+#' @param data population data as data.frame
+#' @export
+#' @import ospsuite
+updateTargetValues <- function(values, targetValues, sourceExpressions, data) {
+  validateIsSameLength(targetValues, sourceExpressions)
+  validateIsOfType(data, "data.frame")
+  validateIsOfLength(values, nrow(data))
+
+  for (sourceIndex in seq_along(sourceExpressions)) {
+    sourceFilter <- eval(sourceExpressions[sourceIndex])
+
+    values[sourceFilter] <- targetValues[sourceIndex]
+  }
+
+  return(values)
+}
+
+#' @title loadStudyDesign
+#' @description Load a StudyDesign object from a file containing a study design table
+#' In this table, Line 1 is path,  Line 2 is unit, Line 3 is type and subsequent lines are values.
+#' The cells for type can include "SOURCE" OR "TARGET". "SOURCE" type can also include subtype "MIN", "MAX" or "EQUALS".
+#' In the current version, unit is expected to be intern units of simulation.
+#' @param studyDesignFile file name of study design table
+#' @return A `StudyDesign` object`
+#' @export
+loadStudyDesign <- function(studyDesignFile) {
+  validateIsString(studyDesignFile)
+  designData <- read.csv(studyDesignFile, header = FALSE, stringsAsFactors = FALSE)
+

```

Comment:
you can safely remove all those spaces here between lines. There is no if etc.. so it should be all without extra space